### PR TITLE
fix: report metrics for stopped machines

### LINKED ISSUE #4224: Missing CPU/disk/memory info on secondary machines
### Bug description

Issue #3111 changed the CPU/disk/memory for Podman machines to Donuts. This looks great on the primary machine, but when you have multiple machines the CPU/disk/memory which used to be grey are now missing. (not entirely sure that's due to #3111, but seems likely?) Not only would it be nice to still have this info on stopped machines, but the missing components lead to layout issues.

Screenshot 2023-10-05 at 3 40 46 PM

### Operating system

macOS 14.0

### Installation Method

None

### Version

next (development version)

### Steps to reproduce

_No response_

### Relevant log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jeffmaury**
Fixes #4224

### What does this PR do?

Reports machine info in donuts when machine is stopped

### What issues does this PR fix or reference?

Fixes #4224

### How to test this PR?

1. Create a machine (do not ask to start it)
2. Go to the resources pages
3. There should be donuts for cpu, memory and disk

### CODE REVIEW

File: extensions/podman/src/extension.ts [hunk 1]
```diff
@@ -169,9 +169,9 @@ async function updateMachines(provider: extensionApi.Provider): Promise<void> {
     const userModeNetworking = isWindows() ? machine.UserModeNetworking : true;
     podmanMachinesInfo.set(machine.Name, {
       name: machine.Name,
-      memory: machineInfo?.memory,
-      cpus: machineInfo?.cpus,
-      diskSize: machineInfo?.diskSize,
+      memory: machineInfo ? machineInfo?.memory : Number(machine.Memory),
+      cpus: machineInfo ? machineInfo?.cpus : machine.CPUs,
+      diskSize: machineInfo ? machineInfo?.diskSize : Number(machine.DiskSize),
```
Comment (lstocchi) [inline extensions/podman/src/extension.ts]: no need for the `?`

**Review by lstocchi — COMMENTED** [review]
I tried and i only see the Disk size for a new created/not started machine.
is that correct or i should also see memory 0 cpu 0?
image

**Comment by lstocchi** [comment]
Tried again, i guess i missed the last commit. Now i see all three values
image

**Comment by lstocchi** [comment]
@jeffmaury i wonder if we can round the value. In my case 677.38 MB goes outside the circle but is it really necessary to know i have .38? Could we just display 677 MB? or round to the first decimal digit, 677.4MB?

**Comment by jeffmaury** [comment]
> @jeffmaury i wonder if we can round the value. In my case 677.38 MB goes outside the circle but is it really necessary to know i have .38? Could we just display 677 MB? or round to the first decimal digit, 677.4MB?

Yes that maybe an option but I don't think this is related to this PR maybe we should have a look at the filesize utility that we use

**Review by lstocchi — APPROVED** [review]
Works fine for me. Just the above remark for the `?` operator

**Comment by deboer-tim** [comment]
> > @jeffmaury i wonder if we can round the value. In my case 677.38 MB goes outside the circle but is it really necessary to know i have .38? Could we just display 677 MB? or round to the first decimal digit, 677.4MB?
> 
> Yes that maybe an option but I don't think this is related to this PR maybe we should have a look at the filesize utility that we use

100% agreed. Since it is a separate issue I opened #4802.

**Comment by benoitf** [comment]
rebased to include https://github.com/containers/podman-desktop/pull/4807

**Pull request merged**